src/parse.c

```diff
@@ -2343,7 +2345,7 @@ handle_arp_ip_targets(NetplanParser* npp, yaml_node_t* node, __unused const void
 
     /* Avoid adding the same arp_ip_targets in a 2nd parsing pass by comparing
      * the array size to the YAML sequence size. Skip if they are equal. */
-    guint item_count = node->data.sequence.items.top - node->data.sequence.items.start;
+    long item_count = node->data.sequence.items.top - node->data.sequence.items.start;
```

Comment:
`top` and `start` are pointers, so `top - start` produces a signed long int. A signed long can be 64bit-long. But I suppose it would be better to use `ptrdiff_t` here.

Reply:
it's still a problem in armhf. `ptrdiff_t` is an `int` there.

```
../src/parse.c: In function ‘handle_arp_ip_targets’:
../src/parse.c:2350:62: error: comparison of integer expressions of different signedness: ‘guint’ {aka ‘unsigned int’} and ‘ptrdiff_t’ {aka ‘int’} [-Werror=sign-compare]
 2350 |     if (npp->current.netdef->bond_params.arp_ip_targets->len == item_count) {
      |                                                              ^~
../src/parse.c: In function ‘handle_wireguard_peers’:
../src/parse.c:2696:51: error: comparison of integer expressions of different signedness: ‘guint’ {aka ‘unsigned int’} and ‘ptrdiff_t’ {aka ‘int’} [-Werror=sign-compare]
 2696 |     if (npp->current.netdef->wireguard_peers->len == item_count) {
      |                                                   ^~
cc1: all warnings being treated as errors
```